doc/rbd/rbd-persistent-write-back-cache.rst

```diff
@@ -0,0 +1,76 @@
+================================
+ RBD Persistent Write-back Cache
+================================
+
+.. index:: Ceph Block Device; Persistent Write-back Cache
+
+Persistent Write-back Cache
+===========================
+
+Persistent Write back cache provides a persistent, fault-tolerant write-back
+cache for Ceph RBD.
+
+This cache is LBA-based, ordered write-back cache which maintains checkpoints
+internally so that writes that get flushed back to the cluster are always
+crash consistent. Even if the client cache is lost entirely, the disk image is
+holding a valid file system that looks like it is just a little bit stale.
+
+Currently this cache uses PMEM devices as cache devices, and later SSDs will
+be supported.
+
+Usage
+=====
+
+Persistent Write-back Image Cache manages the cache data in a persistent device.
+It looks for and creates cache files in a configured directory, and then caches
+data in the file.
+
+Persistent Write-back Image Cache can't be enabled without exclusive lock. It
+tries to enable the write-back cache only when the exclusive lock is got.
+
+It provides two persistence modes. In persistent-on-write mode, the writes
+are completed only when they are persisted to the cache device and will be
+readable after a crash. In persistent-on-flush mode, the writes are completed
+as soon as it no longer needs the callers' data buffer to complete the writes,
+but does not guarantee that writes will be readable after a crash. These data
+are persisted to the cache device when a flush request is received.
+
+Initially it is in persistent-on-write mode, and it switches to
+persistent-on-flush mode if a flush request is received.
+
+Enable Persistent Write back Image Cache
+========================================
+
+To enable RBD persistent write-back image cache, the following Ceph settings
+need to enable in the ``[client]`` `section`_ of your ``ceph.conf`` file::
```

Comment:
We generally try to avoid saying "put things in your ceph.conf" file since we have multiple ways to override configs (ceph.conf, global MON config store, per-pool, per-image, etc). These two settings seem like they would be configured per-image. The settings below are more likely to be configured per-host (perhaps via the ceph.conf file). In general, I think this section needs some cleaning up.

Reply:
Updated.